Add email/pw auth, incl Amplitude + LogRocket

jwayne opened the pull request:
Demo: https://github.com/RomanEngeler1805/LegalResearchAssistant/assets/1303177/6b88a91a-d587-4202-9324-4958be536457

RomanEngeler1805 commented:
Looks good and works smooth to sign-up and login! A few points:

- I would suggest to implement an email / password recovery option if that's possible. I think it's a key feature to a user management system
- Is the choice deliberate to have sign-up and login on the same page? a couple of months ago, I did a tutorial about building a [study app](https://github.com/divanov11/StudyBud/blob/master/base/views.py), and they also had a login and sign-up functionality, might be useful as reference
- for the "Login" button color, adhere to the color palette specified in the [designs](https://www.figma.com/file/ws6r1jr3u3j1Xrz37hTtp9/Legal-Search?node-id=1456%3A61022&mode=dev) i.e the grey seen in the screenshot below
- how hard would it be to show the email next to the user icon as specified in the design? If it's a minimal effort (<30 mins), I would suggest doing it. Additionally, if it's minimal effort, also ask the user for the name in the sign-up process, such that we can personalise the experience by displaying their name as well
![image](https://github.com/RomanEngeler1805/LegalResearchAssistant/assets/43938520/4ebcb3bd-e577-4a06-9a63-e92c3a240b38)

For a later stage, we want to design the login page as in our design
![image](https://github.com/RomanEngeler1805/LegalResearchAssistant/assets/43938520/82b2f529-5a17-4c76-9b0c-6be799243a7a)

RomanEngeler1805 commented on frontend/src/auth/Login.tsx:
```diff
@@ -0,0 +1,258 @@
+            })
+            .catch((error) => {
+              setError(
+                'Es ist ein Fehler aufgetreten. Bitte versuche es erneut.',
```
if the error is that the email is in the DB but the password is wrong, can we provide a more helpful error message? i.e. "Die Email Adresse und das Passwort stimmen nicht überein"

RomanEngeler1805 commented on frontend/src/auth/LoginEmailLinkComplete.tsx:
```diff
@@ -0,0 +1,58 @@
+      if (!email) {
+        // User opened the link on a different device. To prevent session fixation
+        // attacks, ask the user to provide the associated email again. For example:
+        email = window.prompt('Please provide your email for confirmation')
```
if user facing: "Bitte geb deine Email Adresse zur Bestätigung an"

RomanEngeler1805 commented on frontend/src/auth/LoginEmailLinkComplete.tsx:
```diff
@@ -0,0 +1,58 @@
+        email = window.prompt('Please provide your email for confirmation')
+        if (!email) {
+          // eslint-disable-next-line no-alert
+          window.alert('You did not provide an email address')
```
if user facing: "Keine Email Adresse eingegeben"

jwayne commented:
Hi Roman, I added the following:
- Password reset form
- Separate register form, with input box for name (note: you'll have to re-create the accounts you tested)
- Updated design according to the Figma (note: i didn't include the mosaic, since I wasn't sure what to do if the screen became too small to fit the mosaic)
- Show user name in menu

RomanEngeler1805 commented:
Looks neat! Tested it and works as expected.
I will merge it
